OpenXMLSDK.Engine/Word/ReportEngine/Renders/TemplateModelExtensions.cs

```diff
@@ -36,5 +36,22 @@ public static class TemplateModelExtensions
 
             return templateDefinition.Clone().ChildElements;
         }
+
+        /// <summary>
+        /// Replace a template ID by his context definition
+        /// </summary>
+        /// <param name="templateModel">Template model to update</param>
+        /// <param name="contextToUpdate">Context model containing the template ID</param>
+        public static void ReplaceItem(this TemplateModel templateModel, ContextModel contextContainingTheID)
+        {
+            // Search the template ID to use
+            var templateIdToFindInCurrentContext = templateModel.TemplateId;
+            var templateIdToUse = contextContainingTheID.GetItem<StringModel>(templateIdToFindInCurrentContext)?.Value;
```

Comment:
Please use Exists<StringModel>(...), and Get instead Get?.Value and if( != null)